src/agreement/mod.rs

```diff
@@ -466,7 +531,10 @@ impl<NodeUid: Clone + Debug + Ord> Agreement<NodeUid> {
             *self.netinfo.node_index(&self.proposer_id).unwrap(),
             self.epoch,
         );
+        // TODO: Don't spend time creating a `CommonCoin` instance in epochs where the common coin
+        // is known.
         self.common_coin = CommonCoin::new(self.netinfo.clone(), nonce);
+        self.coin_schedule = self.coin_schedule();
```

Comment:
Do we need the coin schedule as a member variable at all? Isn't it easier to just call `self.coin_schedule()` where it's needed?

Reply:
I didn't want to make a call on every `Aux` message.